os/kernel/binary_manager/binary_manager.c

```diff
@@ -140,16 +138,18 @@ int binary_manager(int argc, char *argv[])
 		return 0;
 	}
 
-	/* Execute loading thread for load all binaries */
-	loading_data[0] = itoa(LOADCMD_LOAD_ALL, type_str, 10);
-	loading_data[1] = NULL;
+	/* Create loading thread and initialize resources for loading thread */
+	ret = binary_manager_initialize_loading();
+	if (ret != OK) {
+		bmdbg("Failed to initialize loading, ret %d\n", ret);
+		goto binary_manager_exit;
+	}
 
-	ret = binary_manager_loading(loading_data);
-	if (ret <= 0) {
-		bmdbg("Failed to create loading thread\n");
+	/* Request loading all binaries which would be processed loading thread */
+	ret = binary_manager_loading(LOADCMD_LOAD_ALL, NULL);
+	if (ret != OK) {
+		bmdbg("Failed to request loading all, ret %d\n", ret);
```

Comment:
In this case, binary manager will not be created, so loading_thread should be also killed.

Reply:
Right, I will update it.